src/library/features/crates/cratefeature.cpp

```diff
@@ -793,17 +863,8 @@ void CrateFeature::slotExportTrackFiles() {
 }
 
 void CrateFeature::slotCrateTableChanged(CrateId crateId) {
-    if (m_lastRightClickedIndex.isValid() &&
-            (crateIdFromIndex(m_lastRightClickedIndex) == crateId)) {
-        // Preserve crate selection
-        m_lastRightClickedIndex = rebuildChildModel(crateId);
-        if (m_lastRightClickedIndex.isValid()) {
-            activateCrate(crateId);
-        }
-    } else {
-        // Discard crate selection
-        rebuildChildModel();
-    }
+    Q_UNUSED(crateId);
+    m_pChildModel->reloadTree();
```

Comment:
Rebuilding the tree should preserve the current selection. Otherwise I always need to manually navigate through the hierarchy back to the crate that I had selected.